Mathlib/Algebra/BigOperators/Order.lean

```diff
@@ -830,3 +830,52 @@ theorem IsAbsoluteValue.map_prod [CommSemiring R] [Nontrivial R] [LinearOrderedC
 #align is_absolute_value.map_prod IsAbsoluteValue.map_prod
 
 end AbsoluteValue
+
+namespace Mathlib.Meta.Positivity
+open Qq Lean Meta Finset
+
+/-- The `positivity` extension which proves that `∑ i in s, f i` is nonnegative if `f` is, and
+positive if each `f i` is and `s` is nonempty.
+
+Note that this does not do any complicated reasoning. In particular, it does not try to feed in the
+`i ∈ s` hypothesis to local assumptions, and the only ways it can prove `s` is nonempty is if there
+is a local `s.Nonempty` hypothesis or `s = (univ : Finset α)` and `Nonempty α` can be synthesized by
+TC inference. -/
+@[positivity Finset.sum _ _]
+def evalFinsetSum : PositivityExt where eval {u α} zα pα e := do
+  match e with
+  | ~q(@Finset.sum _ $ι $instα $s $f) =>
+    let (lhs, _, (rhs : Q($α))) ← lambdaMetaTelescope f
+    let so : Option Q(Finset.Nonempty $s) ← do -- TODO: It doesn't complain if we make a typo?
+      try
+        let _fi ← synthInstanceQ q(Fintype $ι)
+        let _no ← synthInstanceQ q(Nonempty $ι)
+        match s with
+        | ~q(@univ _ $fi) => pure (some q(Finset.univ_nonempty (α := $ι)))
```

Comment:
This `let` should be inside the `match`.

Reply:
I moved it, but I'm not sure why you say "should". Are both spellings not functionally equivalent?

Reply:
> Are both spellings not functionally equivalent?

No, because the other order avoids doing a pointless typeclass search if the set was not `univ`. Probably adding an explicit annotation to `_no` fixes the error.

Reply:
What do you mean by "adding an explicit annotation to `-no`"? I've tried several things, but none worked.

Reply:
Oh, your problem is maybe that you can't use `return` here, it breaks Qq. Try pure instead?

Reply:
The following fails still:
```
        match s with
        | ~q(@univ _ $fi) => do
          let no : Q(Nonempty $ι) ← synthInstanceQ q(Nonempty $ι)
          pure (some q(@Finset.univ_nonempty $ι $fi $no))
```